### DISCUSSION TITLE
Delete Event

### DISCUSSION THREAD

**Discussion by tessaherself** [YukiGasai/obsidian-google-calendar#206] [Ideas]
I really really really do need 2 Way Sync :)

I understand that it is hard to build and not purpose of the plugin. Probably even harder to build in hindisight now?

But the create Calendar Event from Frontmatter function is sooooo Epic!!! It just needs an Delete Calendar EVent from Frontmatter function so that it allows for a 2 way sync workaround with the Commander Plugin

**Comment by YukiGasai** [comment]
I am always open for new functionality, how would an example front matter look like for deleting an existing event?

**Reply by tessaherself** [reply]
it would have the "event-id" in it.
So that when I want to change an event i would run a commander makro consisting of:
1. calling your delete event from frontmatter in current file command
2. calling your create event from frontmatter in current file command

and voila there is 2 way sync

**Reply by tessaherself** [reply]
Also while talking about frontmatter:

do you have any recommendations how to write the yaml with easier layout, dateformt?
Because right now I think I have to do this:

![image](https://github.com/YukiGasai/obsidian-google-calendar/assets/73672579/0355c479-ebdd-4e09-99fe-a200f8b9020a)

which gets displayed like this

![image](https://github.com/YukiGasai/obsidian-google-calendar/assets/73672579/1934e97a-e42b-4eb8-8a83-de51da75b585)

I would appreciate 2 things:

1. changing the yaml layout so that properties can display it more beautiful like:
```
start.dateTime:
```
instead of 
```
start:
    dateTime:
```

2. allowing custom dateformats. I think in your code it will always just standard parse with moment.js library. But I am german and hate the american format

**Reply by YukiGasai** [reply]
Okay I understand.
Every event note already has an event-id frontmatter field.
You can just create a new command, read the frontmatter of the current file, extract the ID and run the `deleteEvent` function.
For the yaml problem, take a look at [mappings](https://github.com/YukiGasai/obsidian-google-calendar/issues/30#issuecomment-1347394516).
The date issue is not really solvable. You could try to change the moment's default local, but I tried that once and it collided with other plugins.


Should old event IDs be deleted from the frontmatter after the event is deleted?

**Reply by tessaherself** [reply]
- yes deleting the old event ID is probably a good idea. 

- couldnt you ask for a format in the settings of your plugin and then when using moment() just pass that too?
- but yes if that is too much of an issue, forget about the date-format query.

**Reply by YukiGasai** [reply]
Okay I will add the delete function first, let's turn this into an issue and if everything works we can create an issue for the German date formatting. I think the plugin assumes the English format in a lot of places, with could lead to a lot of problems.

**Reply by tessaherself** [reply]
Wow aweome. It really excites me that I am suddenly involved in a productive Github discussion. This is my first time ever really involving on github with others! Let me know if you want me to help you!



Any format but the american / machine like are fine for me hahaha :)
